### PR TITLE
cli: add deploy option (close #7349)

### LINKED ISSUE #7349: cli: add `hasura deploy` command
Add a CLI command which can do the following operations in one go. 

Config v3 
```
$ hasura metadata apply
$ hasura migrate apply --all-databases
$ hasura metadata reload
```

Config v2
```
$ hasura migrate apply
$ hasura metadata apply
```

### PR DESCRIPTION
**Pull request by atb00ker**
### Description

Add `deploy` option in the cli to which performs the following actions:

```
$ hasura metadata apply
$ hasura migrate apply --all-databases
$ hasura metadata reload
```

### Changelog

- [X] `CHANGELOG.md` is updated with user-facing content relevant to this PR. If no changelog is required, then add the `no-changelog-required` label.

### Affected components

- [X] CLI
- [x] Docs

### Related Issues:
- #7349

### Solution and Design

We are invoking the `Run()` functions of CLI options from `metadata` 
and `migrate` options and performing metadata apply, migration apply and 
metadata reload actions in order.

### Steps to test and verify
Run: `hasura deploy --all-databases` and it should perform:
```
hasura metadata apply
hasura migrate apply --all-databases
hasura metadata reload
```

#### Catalog upgrade
Does this PR change Hasura Catalog version?
- [X] No

#### Metadata
Does this PR add a new Metadata feature?
- [X] No

#### GraphQL
- [X] No new GraphQL schema is generated

#### Breaking changes
- [X] No Breaking changes

### CODE REVIEW

**Comment by atb00ker** [comment]
Hi,

I am in the middle of testing, checking if I need to add docs & writing unit tests.
Just opening a PR for early feedback since I am new here! :smile:

File: cli/commands/migrate.go [hunk 1]
```diff
@@ -43,6 +45,20 @@ func NewMigrateCmd(ec *cli.ExecutionContext) *cobra.Command {
...
+		newMigrateStatusCmd(ec),
+		newMigrateCreateCmd(ec),
+		newMigrateSquashCmd(ec),
+		newMigrateDeleteCmd(ec),
+	)
+
+	return migrateCmd
+}
+
+func migrateCommonFlags(f *pflag.FlagSet, v *viper.Viper) {
```
Comment (purush7) [inline cli/commands/migrate.go:61]:
```suggestion
func migrateCommonFlags(ec *cli.ExecutionContext, f *pflag.FlagSet, v *viper.Viper) {
```

Comment (purush7) [inline cli/commands/migrate.go:61]:
```suggestion
func migrateCommonFlags(ecp cli.ExecutionContext, f *pflag.FlagSet, v *viper.Viper) {
f.StringVar(ecp.Source.Name, "database-name", "", "database on which operation should be applied")
 f.String("endpoint", "", "http(s) endpoint for Hasura GraphQL engine")
 f.BoolVar(ecp.DisableAutoStateMigration, "disable-auto-state-migration", false, "after a config v3 update, disable automatically moving state from hdb_catalog.schema_migrations to catalog state")
```

File: cli/commands/migrate.go [hunk 2]
```diff
@@ -43,6 +45,20 @@ func NewMigrateCmd(ec *cli.ExecutionContext) *cobra.Command {
 	}
 
 	f := migrateCmd.PersistentFlags()
+	migrateCommonFlags(f, v)
```
Comment (purush7) [inline cli/commands/migrate.go:48]:
```suggestion
	migrateCommonFlags(&ec,f, v)
```

File: cli/integration_test/v3/deploy.go [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+		copyMetadataFolder     string
+		expectedMetadataFolder string
+	}{
+		{"apply-all-metadata-and-migrations", &commands.DeployOptions{
+			EC: ec,
+			MetadataApplyOpts: commands.MetadataApplyOptions{
+				EC: ec,
+			},
+			MigrateApplyOpts: commands.MigrateApplyOptions{
+				EC: ec,
```
Comment (purush7) [inline cli/integration_test/v3/deploy.go:48]:
```suggestion
				EC: ec,
				AllDatabases: true,
```

File: docs/graphql/core/hasura-cli/hasura_deploy.rst [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+~~~~~~~~
+
+::
+
+    # Apply Hasura GraphQL engine metadata present in metadata.[yaml|json] file,
+    # run all migrations for all the databases and reload metadata:
+    hasura deploy
+
+    # Use with admin secret:
+    hasura metadata apply --admin-secret "<admin-secret>"
```
Comment (purush7) [inline docs/graphql/core/hasura-cli/hasura_deploy.rst:32]:
```suggestion
    hasura deploy --admin-secret "<admin-secret>"
```

File: docs/graphql/core/hasura-cli/hasura_deploy.rst [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+
+    # Apply metadata to an instance specified by the flag:
+    hasura metadata apply --endpoint "<endpoint>"
```
Comment (purush7) [inline docs/graphql/core/hasura-cli/hasura_deploy.rst:35]:
```suggestion
    hasura deploy --endpoint "<endpoint>"
```

**Review by purush7 — CHANGES_REQUESTED** [review]
Everything is good except those suggested

**Comment by atb00ker** [comment]
@purush7 
I got the first point, but regarding using `ecp`; while I can do that, I must point out, it's inconsistent with the other files; I am assuming ecp stands for execution context pointer; but the pointers in the other files are called `ec` only.

Assuming that the old code had this unintentionally done, I'll update the code in this file.
Just wanted to highlight this, thanks! :smile:

**Comment by atb00ker** [comment]
@purush7  Done! 😄

**Comment by purush7** [comment]
Thanks, @atb00ker for your contribution

**Pull request closed without merging**